cdap-source-control/src/test/java/io/cdap/cdap/sourcecontrol/operationrunner/RemoteSourceControlOperationRunnerTest.java

```diff
@@ -0,0 +1,240 @@
+    cConf = CConfiguration.create();
+    cConf.setInt(Constants.TaskWorker.CONTAINER_KILL_AFTER_REQUEST_COUNT, 0);
+
+    metricsCollectionService = new NoOpMetricsCollectionService();
+    InMemoryDiscoveryService discoveryService = new InMemoryDiscoveryService();
+    remoteClientFactory = new RemoteClientFactory(discoveryService,
+                                                  new DefaultInternalAuthenticator(new AuthenticationTestContext()));
+
+    sConf = SConfiguration.create();
+    cConf.setBoolean(Constants.Security.SSL.INTERNAL_ENABLED, true);
+    cConf.set(Constants.Security.Store.FILE_PATH, SECURE_TEMP_FOLDER.newFolder().getAbsolutePath());
+    sConf.set(Constants.Security.Store.FILE_PASSWORD, "secret");
+    InMemoryNamespaceAdmin namespaceClient = new InMemoryNamespaceAdmin();
+
+    NamespaceMeta namespaceMeta = new NamespaceMeta.Builder().setName(NAMESPACE1).build();
+    namespaceClient.create(namespaceMeta);
+
+    FileSecureStoreService fileSecureStoreService = new FileSecureStoreService(cConf, sConf, namespaceClient,
+                                                                               FileSecureStoreService.CURRENT_CODEC
+                                                                                 .newInstance());
+    
+    httpService = new CommonNettyHttpServiceBuilder(cConf, "test", new NoOpMetricsCollectionService())
+      .setHttpHandlers(
+        new TaskWorkerHttpHandlerInternal(cConf, discoveryService, discoveryService, className -> {
+        }, new NoOpMetricsCollectionService()),
+        new SecureStoreHandler(fileSecureStoreService, fileSecureStoreService)
+      )
+      .setChannelPipelineModifier(new ChannelPipelineModifier() {
+        @Override
+        public void modify(ChannelPipeline pipeline) {
+          pipeline.addAfter("compressor", "decompressor", new HttpContentDecompressor());
+        }
+      })
+      .setExceptionHandler(new HttpExceptionHandler())
+      .build();
+
+    // Start the http service that runs TaskWorkerHttpHandler and register services
+    httpService.start();
+    discoveryService.register(URIScheme.createDiscoverable(Constants.Service.TASK_WORKER, httpService));
+    discoveryService.register(URIScheme.createDiscoverable(Constants.Service.SECURE_STORE_SERVICE, httpService));
+
+    // Set up the git token
+    remoteSecureStore = Guice.createInjector(new AbstractModule() {
+      @Override
+      protected void configure() {
+        bind(RemoteClientFactory.class).toInstance(remoteClientFactory);
+      }
+    }).getInstance(RemoteSecureStore.class);
+    remoteSecureStore.put(NAMESPACE1, TOKEN_NAME, MOCK_TOKEN, "fake git token", ImmutableMap.of("prop1", "value1"));
+  }
+
+  @Before
+  public void beforeEach() throws Exception {
+    cConf.setInt(Constants.SourceControlManagement.GIT_COMMAND_TIMEOUT_SECONDS, GIT_COMMAND_TIMEOUT);
+    cConf.set(Constants.SourceControlManagement.GIT_REPOSITORIES_CLONE_DIRECTORY_PATH,
```

Comment:
why do we need this, I thought each clone happens in it's own directory within this base directory?

Reply:
You're correct, each clone happens in it's own directory. Just that in test, we wanted it to clone in a different root directory in the test `/target`, instead of the default one as `/tmp/source-control/` in cdap-default.xml

Reply:
Also moved to `@BeforeClass`